python/ray/autoscaler/autoscaler.py

```diff
@@ -169,19 +171,23 @@ def _update(self):
         if nodes_to_terminate:
             self.provider.terminate_nodes(nodes_to_terminate)
             nodes = self.workers()
+            managed_workers = self.managed_workers()
             self.log_info_string(nodes, target_workers)
 
         # Terminate nodes if there are too many
         nodes_to_terminate = []
-        while len(nodes) > self.config["max_workers"]:
+        num_to_terminate = max(0, len(nodes) - self.config["max_workers"])
+        while (len(nodes) - len(nodes_to_terminate)
+               ) > self.config["max_workers"] and managed_workers:
+            to_terminate = managed_workers.pop()
             logger.info("StandardAutoscaler: "
-                        "{}: Terminating unneeded node".format(nodes[-1]))
-            nodes_to_terminate.append(nodes[-1])
-            nodes = nodes[:-1]
+                        "{}: Terminating unneeded node".format(to_terminate))
+            nodes_to_terminate.append(to_terminate)
 
         if nodes_to_terminate:
             self.provider.terminate_nodes(nodes_to_terminate)
             nodes = self.workers()
+            managed_workers = self.managed_workers()
```

Comment:
This is doubling the number of API calls to the cloud provider. Instead of doing this, can you:
(1) Just completely ignore unmanaged workers here.
(2) Only ever query unmanaged workers for the purpose of the load metric pruning.